src/cplscheme/CouplingScheme.hpp

```diff
@@ -98,7 +100,54 @@ class CouplingScheme {
    *
    * Does not necessarily advance in time.
    */
-  virtual void advance() = 0;
+  void advance()
+  {
+    firstSynchronization({});
+    firstExchange();
+    secondSynchronization();
+    secondExchange();
+  }
+
+  using ChangedMeshes = std::vector<MeshID>;
+
+  /** Synchronizes mesh changes with remote participants.
+   *
+   * At this point, both participants may have changed the meshes.
+   * Thus, we need to send local changes and receive remote changes.
+   *
+   * @param[in] changes MeshIDs of locally changed meshes
+   *
+   * @returns MeshIDs of remotely changed meshes.
+   */
+  virtual ChangedMeshes firstSynchronization(const ChangedMeshes &changes) = 0;
+
+  /** Exchanges the first set of data.
+   *
+   * @pre \ref firstSynchronization() was called
+   */
+  virtual void firstExchange() = 0;
+
+  /** Receive mesh changes from remote participants in the second step.
+   *
+   * At this point, the remote participant may have changed the meshes if
+   * is coupled using a serial coupling scheme.
+   * So we only need to receive the remote changes here.
```

Comment:
```suggestion
   * At this point, the remote participant may have changed the meshes if
   * it is coupled using a serial coupling scheme.
   * So we only need to receive the remote changes here.
```
The logical connection between both sentences is not clear.